### ISSUE TITLE
garbled display for 'helloitem' example on ChromeOS (aka Android 7.1 on x86) Pixelbook

### ISSUE THREAD

**Issue by NielsMayer** [QUItCoding/qnanopainter#24]
For both Qt5.9.5 and 5.10.1 Android86 builds on Pixelbook running ChromeOS 65, when displayed fullscreen, damage can be seen in display.
![screenshot 2018-04-29 at 12 44 01 pm](https://user-images.githubusercontent.com/6306083/39414980-b651256e-4bf2-11e8-9f62-d8798ac82839.png)

Displayed normal size (1098x618) damage is not seen.
![screenshot 2018-04-29 at 12 44 48 pm](https://user-images.githubusercontent.com/6306083/39414996-d273bec8-4bf2-11e8-84f5-d6d4e560b93d.png)

In both cases the app needs resizing or narrowing so that the bottom of the circle doesn't appear off-screen. However, at least on this stock/default/normal version of ChromeOS on x86 Pixelbook I am unable to free-resize Qt apps between their default size and fullscreen, unlike some android N apps. 

The fact that the circle is offscreen is a separate bug. This bug is about the visible damage seen when the 'helloimage' example is displayed fullscreen.

**Comment by QUItCoding** [comment]
Above patch improves hello* examples and keeps painting inside the window. Also updated latest NanoVG in case that fixes some issues, but didn't see there any changes related to fonts.

Anyway, please report how it looks now on Pixelbook, on my Dell XPS 15 it's like this but I don't have 4k display just HD:
![qnanopainter_helloitem_xps15](https://user-images.githubusercontent.com/16169635/39485049-53c1b3b6-4d80-11e8-93a5-3b2bc65dd460.jpg)

**Comment by NielsMayer** [comment]
Testing again with your latest changes to the examples, and the problem no longer occurs and the images are correctly sized when fullscreen. (Fullscreen, btw, is how apps are auto-sized on pixelbook when in "tablet" mode).

PS: in the new 'hellowindow' the "heart" is not clipping the image, unlike 'helloitem' and 'hellowidget'. Same behavior also seen in Linux.

**Comment by NielsMayer** [comment]
related suggestion: update screenshot on https://github.com/QUItCoding/qnanopainter/blob/master/README.md to the updated QNanoPainter item example.

**Comment by NielsMayer** [comment]
I'm still having some font and refresh-related issues in my own app displayed on pixelbook, even after the latest changes, reminiscent of the bug in QNanoPainter item example in the first photo above. But since it quickly refreshes, the initially visible  "damage" is overdrawn. Unfortunately in the process of overdrawing, the fonts suddenly go tiny... I need to look more closely at changes you made to the examples, etc because so far I just let 'git' do the updating, did no other changes to my own app, then compiled & ran.

**Comment by NielsMayer** [comment]
Sorry for the late reply...

The issues I was having were due to a "premature optimization" I did where instead of stack-allocating a QNanoFont and setting its size over and over, I cached the font in the subclass of the QNanoQuickItemPainter instance. That code, alongside whatever updates had been done since Nov 2017 -- probably the updated nanopainter  -- ended up breaking when I pulled the latest updates. Fortunately, this was the first thing I checked/fixed in my own code, before even beginning to debug, and the problem of the fonts going tiny was instantly solved.

Using "the normal" pattern for fonts with QNanoPainter, my app no longer has the font problem...
e.g., 

```
	QNanoFont rulerFont(QNanoFont::DEFAULT_FONT_THIN_ITALIC);
	rulerFont.setPixelSize(h*0.75f);
	p->setFont(rulerFont);
```

This issue can thus be closed.
